Temporarily add back $zone and $endpoint.

kwisatz opened the pull request:
Re-add deprecated $zone and $endpoint parameters to allow for a transition period, especially for users using virtual or exported resources.
In fact, with exported resources in storeconfigs that use the old parameters, all puppet runs will fail when upgrading the module to a version
that only has the new parameters. Having a transition period will allow puppet to run and export new resources to storeconfig that use the new parameter names.

xaque208 commented on manifests/params.pp:
```diff
@@ -21,9 +21,6 @@
   $install_nagios_plugins = true
 
-  #whether to install packages that provide the 'mail' binary
-  $install_mail_utils_package = false
```
This isn't a documentation change as the commit suggests.  What is this change here to address?

lbischof replied:
It seems to be leftover code from a rename of the parameter. It doesn't get used anywhere else.

xaque208 replied:
This change should be a separate commit in my opinion.

lbischof replied:
I agree. I'll seperate it, when I'm at a computer.

On Tue, 3 May 2016, 21:52 Zach Leslie, <notifications@github.com> wrote:

> In manifests/params.pp
> <https://github.com/Icinga/puppet-icinga2/pull/204#discussion_r61944309>:
>
> > @@ -21,9 +21,6 @@
> >    #Whether to install the plugin packages when the icinga2 class is applied:
> >    $install_nagios_plugins = true
> >
> > -  #whether to install packages that provide the 'mail' binary
> > -  $install_mail_utils_package = false
>
> This change should be a separate commit in my opinion.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/Icinga/puppet-icinga2/pull/204/files/68275a92606bdeb4e3bdc1891bb114be42c26163#r61944309>
>

lbischof commented:
Ok I seperated my changes into two commits.

xaque208 commented:
+1 Would merge.

kwisatz committed:
Temporarily add back $zone and $endpoint.

Re-add deprecated $zone and $endpoint parameters to allow for a transition period, especially for users using virtual or exported resources.
In fact, with exported resources in storeconfigs that use the old parameters, all puppet runs will fail when upgrading the module to a version
that only has the new parameters. Having a transition period will allow puppet to run and export new resources to storeconfig that use the new parameter names.

bobapple commented:
Thank you @kwisatz 
As always, I appreciate your contribution to the module. I would like to skip this PR. I'm sorry for the inconvenience the change caused in your environment. Backwards compatibility is important, however the changes were made in an early development version.

kwisatz commented:
@bobapple no worries. I wouldn't have created this if it wasn't for @lazyfrosch asking me to do so.
